src/test/java/springboot/webapp/usersmanager/controllers_tests/UserControllerTests.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Test
+    public void getByIdWhenNonExistentUserStatusNotFound() throws Exception {
+        when(userService.get(1)).thenReturn(Optional.empty());
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                .get("/users/2")
+                .contentType(MediaType.APPLICATION_JSON)).andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        MatcherAssert.assertThat(response.getStatus(), is(equalTo(HttpStatus.NOT_FOUND.value())));
+    }
+
+    @Test
+    public void putWhenNonExistentUserStatusOk() throws Exception {
+
+        when(userService.put(getUser())).thenReturn(true);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                .put("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(this.mapper.writeValueAsString(getUser()))).andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        MatcherAssert.assertThat(response.getStatus(), is(equalTo(HttpStatus.OK.value())));
+    }
+
+    @Test
+    public void putWhenUserWithExistedEmailStatusConflict() throws Exception {
+        when(userService.put(any())).thenThrow(IllegalArgumentException.class);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                .put("/users")
+                .accept(MediaType.APPLICATION_JSON)
+                .content(this.mapper.writeValueAsString(getUser()))
+                .contentType(MediaType.APPLICATION_JSON)).andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        MatcherAssert.assertThat(response.getStatus(), is(equalTo(HttpStatus.CONFLICT.value())));
+    }
+
+    @Test
+    public void deleteWhenExistedUserStatusOk() throws Exception {
+        when(userService.delete(user2.getId())).thenReturn(true);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                        .delete("/users/" + user2.getId())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        MatcherAssert.assertThat(response.getStatus(), is(equalTo(HttpStatus.OK.value())));
+    }
+
+    @Test
+    public void deleteByIdWhenNonExistentUserStatusNotFound() throws Exception {
+        when(userService.delete(6)).thenReturn(true);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                        .delete("/users/7")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        MatcherAssert.assertThat(response.getStatus(), is(equalTo(HttpStatus.NOT_FOUND.value())));
+    }
+
+    public User getUser() {
+        PodamFactory factory = new PodamFactoryImpl();
+        return factory.manufacturePojo(User.class);
```

Comment:
for String UUID.randomUUID().toString()
for numbers java.util.Random;

Reply:
+